packages/SwingSet/test/upgrade/upgrade.test.js

```diff
@@ -836,91 +838,100 @@ test('failed upgrade - lost promise export', async t => {
       puppet: '../../tools/vat-puppet.js',
     },
   });
-  const { messageToVat, messageToObject } = await initKernelForTest(
-    t,
-    t.context.data,
-    config,
-  );
-  // Open-coded `EV` analog.
-  const bootstrapHandler = {
-    get: (_target, key, _receiver) => {
-      const sendMessage = (...args) => messageToVat('bootstrap', key, ...args);
-      return sendMessage;
-    },
-  };
-  const EBootstrap = new Proxy({}, bootstrapHandler);
+  const { controller } = await initKernelForTest(t, t.context.data, config);
+  const { EV } = makeRunUtils(controller);
 
-  // Create spies in the bootstrap vat.
-  const bootstrapSpy = await EBootstrap.makeRemotable('Spy', {
+  // Create a spy in the bootstrap vat.
+  const spy = await EV.vat('bootstrap').makeRemotable('Spy', {
     log: true,
     onFulfilled: true,
     onRejected: true,
   });
   const expectSpyCalls = [];
   const assertSpyCalls = async (message, newCalls) => {
     expectSpyCalls.push(...newCalls);
-    const actualSpyCalls = await EBootstrap.getLogForRemotable(bootstrapSpy);
-    arrayIsLike(t, actualSpyCalls, expectSpyCalls, message);
+    const actualSpyCalls = await EV.vat('bootstrap').getLogForRemotable(spy);
+
+    // Where we expect a promise and there is actually a promise in the right
+    // position, update our expectation to that exact instance.
+    const expectation = [];
+    for (let i = 0; i < expectSpyCalls.length; i += 1) {
+      const expectCall = expectSpyCalls[i];
+      const [expectMethodName, ...expectArgs] = expectCall;
+      const [actualMethodName, ...actualArgs] = actualSpyCalls[i] || [];
+      if (actualMethodName !== expectMethodName) {
+        expectation.push(expectCall);
+        continue;
+      }
+      const updatedExpectArgs = expectArgs.map((arg, i) =>
+        isPromise(arg) && isPromise(actualArgs[i]) ? actualArgs[i] : arg,
+      );
```

Comment:
I'm not sure I understand why we're doing this. This is effectively saying any promise in the right position will work, right, even if the 2 promises are unrelated (because we can't test promise identity?).

I would kinda prefer using an explicit sentinel value, maybe a unique symbol?